scene/gui/label.cpp

```diff
@@ -447,12 +452,20 @@ void Label::regenerate_word_cache() {
 
 			if (current == '\n') {
 				insert_newline = true;
+				prev_insert_newline = true;
 			} else if (current != ' ') {
 				total_char_cache++;
 			}
 
 			if (i < xl_text.length() && xl_text[i] == ' ') {
-				if (line_width > 0 || last == nullptr || last->char_pos != WordCache::CHAR_WRAPLINE) {
+				if (i == 0 || (i != 0 && prev_insert_newline)) {
```

Comment:
```suggestion
				if (i == 0 || prev_insert_newline) {
```

It's redundant here I believe.

Also, could the `i == 0` check be removed here, and make `prev_insert_newline` default to `true` instead?

Reply:
I don't think you can remove the i==0 since the very first original issue was caused by the fact that, the case when xl_text[i] when i = 0 (which is the very first character of the string) is equal to the space character, is missing